.github/workflows/library_dafny_verification.yml

```
@@ -21,7 +21,7 @@ jobs:
         library: [
           AwsEncryptionSDK
         ]
-        os: [ macos-latest ]
+        os: [ macos-12 ]
```

Verification can run on latest, no crypto involved.
```suggestion
        os: [ macos-latest ]
```

---

We only need macos-12 for .Net related checks. I changed all the macos-latest to macos-12 because other repo (like db-esdk) had made similar change and wanted to remain consistent with our other repos.